src/Microsoft.Data/DataFrameTable.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrameTable is just a container that holds a number of DataFrameColumns. It mainly acts as a convenient store to allow DataFrame to implement its algorithms
+    /// </summary>
+    internal class DataFrameTable
+    {
+        private IList<BaseDataFrameColumn> _columns;
+
+        private List<string> _columnNames = new List<string>();
+        private Dictionary<string, int> _columnNameToIndexDictionary = new Dictionary<string, int>();
+
+        public long RowCount { get; private set; } = 0;
+        public int ColumnCount { get; private set; } = 0;
+        public DataFrameTable()
+        {
+            _columns = new List<BaseDataFrameColumn>();
+        }
+
+        public DataFrameTable(IList<BaseDataFrameColumn> columns)
+        {
+            columns = columns ?? throw new ArgumentNullException(nameof(columns));
+            _columns = columns;
```

Comment:
I don't know what the scenario is for this ctor, but it should be such that the caller has full knowledge that you're going to take ownership of the list and they presumably are to not touch it again. In general it is a bit unusual for a non trivial type to be passed a field to use directly as a private field. It's not really very private. But of course depends on the semantics you're looking for